### DISCUSSION TITLE
Return object full of data when insert

### DISCUSSION THREAD

**Discussion by dvrg** [encode/databases#412] [Q&A] [answered]
Hi, i have questions. In old sqlalchemy query, after insert we can have return object full of data what we insert, but in this query only return the id of row. How i can get return object with full of data ?

```python
@database.transaction()
async def create(self, *, obj_in: DeviceCreate) -> Device:
    obj_in_data = jsonable_encoder(obj_in)
    installed_app = obj_in_data.pop("installed_applications")
    query = insert(self.model).values(
        **obj_in_data, installed_applications=json.dumps(installed_app)
    )
    return await database.execute(query=query)

# return --> 6 (id of row)
```

**Comment by aminalaee** [comment, marked as answer]
Hi @dvrg 
As far as I know this is the expected behaviour. You need a separate query to select all columns with the returned id.

I think what you want to do is beyond what a driver like `databases` will do. You need to use an ORM like [orm](https://github.com/encode/orm) to do that for you.

**Reply by dvrg** [reply]
i think do same as you suggests before, but i not sure. Thankyou for clarify the solutions.
